File: Formação Lógica de Programação/Gerenciamento de Itens Mágicos.js
// Desafios JavaScript na DIO têm funções "gets" e "print" acessíveis globalmente:
// - "gets" : lê UMA linha com dado(s) de entrada (inputs) do usuário;
// - "print": imprime um texto de saída (output), pulando linha.

class ItemMagico {
  constructor(tipo, dano, resistencia) {
    this.tipo = tipo;
    this.dano = dano;
    this.resistencia = resistencia;
  }

  calcularDano() {
    return this.tipo === 'arma' ? this.dano * 2 : this.dano;
  }
}

// (Valores simulados para teste local)
const tipoItem = "arma";         // No desafio será: gets();
const danoItem = 50;             // No desafio será: parseInt(gets());
const resistenciaItem = 100;     // No desafio será: parseInt(gets());

const itemPersonalizado = new ItemMagico(tipoItem, danoItem, resistenciaItem);

// Imprima os atributos do item personalizado (substituindo print por console.log)
console.log("Tipo: " + itemPersonalizado.tipo);
console.log("Dano: " + itemPersonalizado.dano);
console.log("Resistencia: " + itemPersonalizado.resistencia);

const danoTotal = itemPersonalizado.calcularDano();
console.log("Dano em combate: " + danoTotal);

File: Formação Lógica de Programação/Python/Gerenciamento de Itens Mágicos.py
# Desafios Python na DIO geralmente usam a função "input" para ler dados 
# e a função nativa "print" para imprimir a saída.

class ItemMagico:
    def __init__(self, tipo, dano, resistencia):
        self.tipo = tipo
        self.dano = dano
        self.resistencia = resistencia

    def calcular_dano(self):
        # [resultado_verdadeiro] if [condicao] else [resultado_falso]
        return self.dano * 2 if self.tipo == 'arma' else self.dano


tipo_item = "arma"          # No desafio será: input()
dano_item = 50              # No desafio será: int(input())
resistencia_item = 100      # No desafio será: int(input())

item_personalizado = ItemMagico(tipo_item, dano_item, resistencia_item)

print(f"Tipo: {item_personalizado.tipo}")
print(f"Dano: {item_personalizado.dano}")
print(f"Resistencia: {item_personalizado.resistencia}")

dano_total = item_personalizado.calcular_dano()
print(f"Dano em combate: {dano_total}")